Libraries/xcsdk/Headers/xcsdk/SDK/Manager.h

```diff
@@ -89,6 +89,12 @@ class Manager {
      */
     std::vector<std::string> executablePaths() const;
 
+    /*
+     * Conglomeration of executable paths that optionally includes extra toolchains
+     * and the paths from an SDK target.
+     */
+    std::vector<std::string> executablePaths(std::vector<xcsdk::SDK::Toolchain::shared_ptr> &toolchains, xcsdk::SDK::Target::shared_ptr &target) const;
```

Comment:
I'm not sure if this structure is better for this method than before: I think in most cases it's an error for `target` to be `nullptr` here.

The fact that it's valid in `xcrun` seems like part of the command-line API of that tool, rather than something worth baking into `xcsdk`. With the old method as-is, you'd have something like this:

    if (target != nullptr) {
        executablePaths.extend(target.executablePaths(toolchains));
    } else {
        for (xcsdk::SDK::Toolchain::shared_ptr const &toolchain : toolchains) {
            executablePaths.extend(toolchain.executablePaths());
    }

I don't think that's so bad?